Issue/wc product model updates

nbradbury opened the pull request:
This updates the `WCProductModel` and related table with a few fields we missed the first time around:

* Downloadable files
* Download expiry
* Download limit
* External URL
* Related product IDs
* Cross-sell product IDs
* Upsell product IDs

This PR also adds the ability to parse a product's attributes, get a comma-separate list of tags and categories, and delete all products for a site.

AmandaRiu commented on fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java:
```diff
@@ -638,6 +620,66 @@ private void migrateAddOn(@AddOn String addOnName, SQLiteDatabase db, int oldDbV
+                               + "ATTRIBUTES TEXT NOT NULL,"
+                               + "RELATED_IDS TEXT NOT NULL,"
+                               + "CROSS_SELL_IDS TEXT NOT NULL,"
+                               + "UPSELL_IDS TEXT NOT NULL,"
```
Column `UPSELL_IDS` should be `UP_SELL_IDS`, or, to better match the json response, `WCProductModel` should be changed to `upsellIds`

nbradbury replied:
Changed in 40074db

nbradbury commented on fluxc/src/main/java/org/wordpress/android/fluxc/persistence/WellSqlConfig.java:
```diff
@@ -583,27 +584,8 @@ private void migrateAddOn(@AddOn String addOnName, SQLiteDatabase db, int oldDbV
                     break;
                 case 56:
                     AppLog.d(T.DB, "Migrating addon " + addOnName + " to version " + (oldDbVersion + 1));
+                    // redefined in 59 below
```
Reverted in c81530e

nbradbury commented:
> The db version has not been updated...The version specified in the case sequence should also be updated

Looks like this was due to a botched merge from `develop`. The version of the db was correct when I first created this PR, but then `develop` updated it again and I resolved the conflict incorrectly. Fixed in 8e380ce.

>  it would be good to add some tests for the getCommaSeparatedOptions() method

Added in 11db15a. Ready for another round!
